### PR TITLE
overlord/devicestate, daemon: record the seed current system was installed from

### PR DESCRIPTION
**Pull request by bboozzoo**
The PR adds support for recording which seed was used for installing the current system. This allows snapd to properly indicate the *current* recovery system when returning a list over `/v2/systems` endpoint.

Needed for the chooser

### CODE REVIEW

File: overlord/devicestate/firstboot.go [hunk 1]
```diff
@@ -280,6 +281,12 @@ func populateStateFromSeedImpl(st *state.State, opts *populateStateFromSeedOptio
 		endTs.AddTask(preseedDoneTask)
 		markSeeded.WaitFor(preseedDoneTask)
 	}
+	whatSeeds := &seededSystem{
+		System:  sysLabel,
+		Model:   model.Model(),
+		BrandID: model.BrandID(),
```
Comment (pedronis) [inline overlord/devicestate/firstboot.go]: let's add model.Revision() (as "revision")  and model.Timestamp() (as "timestamp")  as well, for better tracking

File: overlord/devicestate/handlers.go [hunk 2]
```diff
@@ -121,6 +140,16 @@ func (m *DeviceManager) doMarkSeeded(t *state.Task, _ *tomb.Tomb) error {
...
+	if err := t.Get("seed-system", &whatSeeded); err != nil && err != state.ErrNoState {
+		return err
+	}
+	if whatSeeded != nil {
+		// TODO:UC20 what about remodels?
+		if err := m.recordSeededSystem(st, whatSeeded); err != nil {
+			return fmt.Errorf("cannot record the seeded system: %v", err)
+		}
+	}
 	st.Set("seed-time", time.Now())
```
Comment (pedronis) [inline overlord/devicestate/handlers.go]: let's get this time early here and set it also as "seed-time" in the seeded-systems entry

File: overlord/devicestate/firstboot.go (hunk 1 above)
Comment (bboozzoo) [reply overlord/devicestate/firstboot.go]: This is done now

File: overlord/devicestate/handlers.go (hunk 2 above)
Comment (bboozzoo) [reply overlord/devicestate/handlers.go]: This one too.

**Pull request merged**